test/EndToEnd/ProjectBuildTests.cs

```diff
@@ -156,58 +139,248 @@ public void ItCanPublishArm64Wpf()
+        /// [Windows only tests]
+        /// The test checks if the template creates the template for correct framework by default.
+        /// For .NET 6 the templates should create the projects targeting net6.0.
+        /// </summary>
+        [WindowsOnlyTheory]
+        [InlineData("wpf")]
+        [InlineData("wpf", "C#")]
+        [InlineData("wpf", "VB")]
+        [InlineData("wpflib")]
+        [InlineData("wpflib", "C#")]
+        [InlineData("wpflib", "VB")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfcustomcontrollib", "C#")]
+        [InlineData("wpfcustomcontrollib", "VB")]
+        [InlineData("wpfusercontrollib")]
+        [InlineData("wpfusercontrollib", "C#")]
+        [InlineData("wpfusercontrollib", "VB")]
+        [InlineData("winforms")]
+        [InlineData("winforms", "C#")]
+        [InlineData("winforms", "VB")]
+        [InlineData("winformslib")]
+        [InlineData("winformslib", "C#")]
+        [InlineData("winformslib", "VB")]
+        [InlineData("winformscontrollib")]
+        [InlineData("winformscontrollib", "C#")]
+        [InlineData("winformscontrollib", "VB")]
+        public void ItCanCreateAndBuildTemplatesWithDefaultFramework_Windows(string templateName, string language = "")
+        {
+            string framework = DetectExpectedDefaultFramework();
+            TestTemplateCreateAndBuild(templateName, selfContained: true, language: language, framework: $"{framework}-windows");
+        }
+
+        /// <summary>
+        /// [project is not built on linux-musl]
+        /// The test checks if the template creates the template for correct framework by default.
+        /// For .NET 6 the templates should create the projects targeting net6.0.
+        /// </summary>
+        [Theory]
+        [InlineData("grpc")]
+        public void ItCanCreateAndBuildTemplatesWithDefaultFramework_DisableBuildOnLinuxMusl(string templateName)
+        {
+            string framework = DetectExpectedDefaultFramework();
+
+            if (RuntimeInformation.RuntimeIdentifier.StartsWith("alpine")) //linux musl
+            {
+                TestTemplateCreateAndBuild(templateName, build: false, framework: framework);
+            }
+            else
+            {
+                TestTemplateCreateAndBuild(templateName, selfContained: true, framework: framework);
+            }
+        }
+
+        private static string DetectExpectedDefaultFramework()
+        {
+            string dotnetFolder = Path.GetDirectoryName(RepoDirectoriesProvider.DotnetUnderTest);
+            string[] runtimeFolders = Directory.GetDirectories(Path.Combine(dotnetFolder, "shared", "Microsoft.NETCore.App"));
+
+            int latestMajorVersion = runtimeFolders.Select(folder => int.Parse(Path.GetFileName(folder).Split('.').First())).Max();
+            if (latestMajorVersion == 6)
+            {
+                return "net6.0";
+            }
+            throw new Exception("Unsupported version of SDK");
```

Comment:
Looking at past releases, feels like it safe bet to just do `"net" + runtimeFolders.Sort().First().Remove(positonOfSecondDot)` which would result in `net5.0`, `net6.0`, `net7.0`, but also `net8.1` if that happens...

Reply:
Yeah, I was thinking about same however default framework is not always latest. For .NET 5.0 the default framework is 3.1. It might be that for .NET 7.0 the default framework will be net6.0.